### PR TITLE
Add Client and Server traits

### PR DESCRIPTION
**Pull request by yoshuawuyts**
Adds `Client` and `Server` traits, replacing `http-service` and `http-client` and providing a bridge between the two traits.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -111,29 +112,36 @@ pub mod url {
...
 mod method;
 mod request;
 mod response;
-mod server;
 mod status;
 mod status_code;
 mod type_map;
 mod version;
 
+cfg_unstable! {
```
Comment (otavio) [inline src/lib.rs:132]: Is this macro worth it? You use it only once.

File: src/lib.rs [hunk 2]
```diff
@@ -95,8 +95,9 @@
 
 #![forbid(rust_2018_idioms)]
 #![deny(missing_debug_implementations, nonstandard_style)]
-#![warn(missing_docs, missing_doc_code_examples, unreachable_pub)]
```
Comment (otavio) [inline src/lib.rs:98]: Why remove this? better would add an example for it.

Comment (yoshuawuyts) [reply src/lib.rs:98]: That'd be several hundred lines, which is out of scope for this PR.

**Comment by yoshuawuyts** [comment]
Merging as this is marked "unstable"; allows us to experiment with it. Thanks!

**Pull request merged**